libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1641,18 +1641,34 @@ AP_GPS_UBLOX::_parse_gps(void)
             MB_Debug("RELPOSNED flags: %lx valid: %lx invalid: %lx\n", _buffer.relposned.flags, valid_mask, invalid_mask);
             if (((_buffer.relposned.flags & valid_mask) == valid_mask) &&
                 ((_buffer.relposned.flags & invalid_mask) == 0)) {
-                if (calculate_moving_base_yaw(_buffer.relposned.relPosHeading * 1e-5,
-                                          _buffer.relposned.relPosLength * 0.01,
-                                          _buffer.relposned.relPosD*0.01)) {
-                    state.have_gps_yaw_accuracy = true;
-                    state.gps_yaw_accuracy = _buffer.relposned.accHeading * 1e-5;
-                    _last_relposned_ms = AP_HAL::millis();
-                }
+
+                // Populte state
                 state.relPosHeading = _buffer.relposned.relPosHeading * 1e-5;
                 state.relPosLength  = _buffer.relposned.relPosLength * 0.01;
                 state.relPosD       = _buffer.relposned.relPosD * 0.01;
                 state.accHeading    = _buffer.relposned.accHeading * 1e-5;
                 state.relposheading_ts = AP_HAL::millis();
+
+                // Add on high presision components if avalable
+                if (_buffer.relposned.relPosHPLength > -100 && _buffer.relposned.relPosHPLength < 100) {
+                    // 0.1mm to m
+                    state.relPosLength += _buffer.relposned.relPosHPLength * 0.0001;
+                }
+                if (_buffer.relposned.relPosHPD > -100 && _buffer.relposned.relPosHPD < 100) {
+                    // 0.1mm to m
+                    state.relPosD += _buffer.relposned.relPosHPD * 0.0001;
+                }
+
+                // un-normalized pos down if necessary
+                if ((_buffer.relposned.flags & uint32_t(RELPOSNED::relPosNormalized)) != 0) {
```

Comment:
the datasheet doesn't say what it is normalized to. 1 meter? 1 cm? pi?
what testing have you done of this? do you have captures showing UBX data where it has set the normalization flag?

Reply:
No testing, just based on reading the datasheet. I have had a deeper look and can't find any more info. I think this is what they probably mean, but in the absence of anything concrete I guess we could also just refuse to accept the message if the flag is set. Just ignoring the flag as we do currently seems like a bad approach.

Reply:
Hum: https://content.u-blox.com/sites/default/files/documents/ZED-F9P-MovingBase_AppNote_UBX-19009093.pdf

page 20.

![image](https://github.com/user-attachments/assets/e531e8ce-df1f-44a0-bae2-51e459a9bba9)

https://www.u-blox.com/sites/default/files/ZED-F9H_IntegrationManual_UBX-19030120.pdf

![image](https://github.com/user-attachments/assets/b9134e46-93c0-42ff-80c6-e3c1608b2bbc)

Reply:
It seems like we would need a F9H and a F9P pair to get this behavior. I don't have the hardware to test that combo.

Reply:
Nobody has heard of the F9H, so we don't need to support it. I have removed this change. We may want to check that this flag is not set in the future.